pkg/timeutil/window.go

```
@@ -0,0 +1,154 @@
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	nilDay = time.Weekday(-1)
+
+	shortDayNames = map[string]time.Weekday{
+		"sun": time.Sunday,
+		"mon": time.Monday,
+		"tue": time.Tuesday,
+		"wed": time.Wednesday,
+		"thu": time.Thursday,
+		"fri": time.Friday,
+		"sat": time.Saturday,
+	}
+)
+
+func d2s(d time.Weekday) string {
+	s := d.String()
+	if len(s) >= 3 {
+		return s[:3]
+	}
+	return s
+}
+
+type dayTime struct {
+	day     time.Weekday // day of week, -1 if unset
+	minutes uint64       // minutes into day
+}
+
+func parseTime(s string) (hh, mm uint64, err error) {
+	spl := strings.Split(s, ":")
+	if len(spl) != 2 {
+		err = errors.New("malformed HH:MM specification")
+		return
+	}
+
+	hh, err = strconv.ParseUint(spl[0], 10, 0)
+	if err != nil {
+		err = fmt.Errorf("malformed hour %q: %s", spl[0], err)
+		return
+	}
+
+	mm, err = strconv.ParseUint(spl[1], 10, 0)
+	if err != nil {
+		err = fmt.Errorf("malformed minute %q: %s", spl[1], err)
+		return
+	}
+
+	return
+}
+
+func parseDayTime(s string) (dayTime, error) {
+	dt := dayTime{nilDay, 0}
+	spl := strings.Fields(s)
+
+	switch len(spl) {
+	default:
+		fallthrough
```

This should return a distinct error.